### PR TITLE
PP-4597 output timestamps in miliseconds

### PR DESCRIPTION
**Pull request by georgeracu**
## WHAT YOU DID
- Removed custom JSON deserializer and using a common one
- Bumped pay-java-common 
- Code cleanup

## How to test

- How should it be reviewed? 
- What feedback do you need? 
- If manual testing is needed, give suggested testing steps.

### Logging

- [ ] only emit log lines at ERROR level which require immediate attention from a support engineer. These will trigger a zendesk alert.

### CODE REVIEW

File: src/main/java/uk/gov/pay/api/service/ConnectorUriGenerator.java [hunk 1]
```diff
@@ -85,11 +85,9 @@ public String eventsURI(Account account, Optional<ZonedDateTime> toDate, Optiona
 
         Map<String, String> params = new LinkedHashMap<>();
 
-        if (toDate.isPresent())
-            params.put("to_date", DateTimeUtils.toUTCDateString(toDate.get()));
+        toDate.ifPresent(zonedDateTime -> params.put("to_date", DateTimeUtils.toUTCDateString(zonedDateTime)));
```
Comment (alexbishop1) [inline src/main/java/uk/gov/pay/api/service/ConnectorUriGenerator.java]: I think we can just use the formatter directly here:

```java
toDate.ifPresent(zonedDateTime -> params.put(ISO_INSTANT_MILLISECOND_PRECISION.format(zonedDateTime));
```

Comment (georgeracu) [reply src/main/java/uk/gov/pay/api/service/ConnectorUriGenerator.java]: That's well spotted. We can remove this wrapper method and use the formatter directly :)

File: src/main/java/uk/gov/pay/api/service/ConnectorUriGenerator.java [hunk 2]
```diff
@@ -85,11 +85,9 @@ public String eventsURI(Account account, Optional<ZonedDateTime> toDate, Optiona
 
         Map<String, String> params = new LinkedHashMap<>();
 
-        if (toDate.isPresent())
-            params.put("to_date", DateTimeUtils.toUTCDateString(toDate.get()));
+        toDate.ifPresent(zonedDateTime -> params.put("to_date", ISO_INSTANT_MILLISECOND_PRECISION.format(zonedDateTime)));
```
Comment (alexbishop1) [inline src/main/java/uk/gov/pay/api/service/ConnectorUriGenerator.java]: Just realised we can do this!

```java
toDate.map(ISO_INSTANT_MILLISECOND_PRECISION::format).ifPresent(timestamp -> params.put("from_date", timestamp));
```

Comment (georgeracu) [reply src/main/java/uk/gov/pay/api/service/ConnectorUriGenerator.java]: 👍 done

**Pull request merged**